api/types/registry/registry.go

```diff
@@ -102,3 +105,14 @@ type SearchResults struct {
 	// Results is a slice containing the actual results for the search
 	Results []SearchResult `json:"results"`
 }
+
+// ManifestInspect describes the result obtained from contacting the
+// registry to retrieve image metadata
+// TODO(nishanttotla): Decide which other fields to add here
+type ManifestInspect struct {
+	// Digest is the content addressable digest for the image on the registry
+	Digest digest.Digest
```

Comment:
What about returning the top-level manifest itself instead of (or in addition to) the digest? I could imagine this being useful for some things.

We could also return an optional config blob if one is referenced by the top-level manifest.

Then the platform determination could be done on the client side, making this `/manifest` more generic.

It's still not really ideal, because the daemon is deciding what items to fetch from the registry and return to the client. I kind of like the idea of proxying the registry API through the daemon, but that's a pretty big project and probably way out of scope for what we're trying to do here.

That said, I don't feel strongly about any of this, and I could be okay with just returning specific fields such as `Platforms` in the API response if that's what people prefer.